src/test/java/uk/gov/pay/directdebit/webhook/gocardless/resources/WebhookGoCardlessResourceMandateActionsIT.java

```diff
@@ -0,0 +1,182 @@
+        assertThat(mandate.get("state_details"), is("mandate_reinstated"));
+        assertThat(mandate.get("state_details_description"), is("A cancelled mandate has been re-instated by the customer's bank."));
+    }
+
+    @Test
+    public void cancelledChangesStateToCancelledAndSendsMandateCancelledEmail() throws IOException {
+        stubEmail(EmailTemplate.MANDATE_CANCELLED);
+
+        postWebhook("gocardless-webhook-mandate-cancelled.json",
+                "b019901d05cedd31ca878b33789010cbd012f579b079af760cf03c275eeb6b13");
+
+        Map<String, Object> mandate = testContext.getDatabaseTestHelper().getMandateById(mandateFixture.getId());
+        assertThat(mandate.get("state"), is(MandateState.CANCELLED.toString()));
+        assertThat(mandate.get("state_details"), is("mandate_cancelled"));
+        assertThat(mandate.get("state_details_description"), is("The mandate was cancelled at a bank branch."));
+
+        verifyEmailSent(EmailTemplate.MANDATE_CANCELLED);
+    }
+
+    @Test
+    public void failedChangesStateToFailedAndSendsMandateFailedEmail() throws IOException {
+        stubEmail(EmailTemplate.MANDATE_FAILED);
+
+        postWebhook("gocardless-webhook-mandate-failed.json",
+                "1978e4564c85b026146649b149d6ebb20e3395d5213b2eba72fd32da98ecfff2");
+
+        Map<String, Object> mandate = testContext.getDatabaseTestHelper().getMandateById(mandateFixture.getId());
+        assertThat(mandate.get("state"), is(MandateState.FAILED.toString()));
+        assertThat(mandate.get("state_details"), is("invalid_bank_details"));
+        assertThat(mandate.get("state_details_description"), is("The specified bank account does not exist or was closed."));
+
+        verifyEmailSent(EmailTemplate.MANDATE_FAILED);
+    }
+
+    @Test
+    @Ignore
+    public void expiredChangesStateToFailed() throws IOException {
+        postWebhook("gocardless-webhook-mandate-expired.json",
+                "492f08e710849470e7103212b8ee4256dc193fa57561488ed6640435684bab39");
+
+        Map<String, Object> mandate = testContext.getDatabaseTestHelper().getMandateById(mandateFixture.getId());
+        assertThat(mandate.get("state"), is(MandateState.EXPIRED.toString()));
+        assertThat(mandate.get("state_details"), is("mandate_expired"));
+        assertThat(mandate.get("state_details_description"), is("The mandate is being marked as expired, because no payments have been collected " +
+                "against it for the dormancy period of your service user number. If you have access to the mandate reinstation API endpoint, you can use " +
+                "this to attempt to set this mandate up again."));
+    }
+
+    private void postWebhook(String webhookBodyResourceName, String webhookSignature) throws IOException {
+        given().port(testContext.getPort())
+                .body(Resources.toString(Resources.getResource(webhookBodyResourceName), UTF_8))
```

Comment:
Was going to suggest you could use Dropwizard's `fixture(webhookBodyResourceName)` (which still uses `UTF_8`) but interestingly that doesn't remove the extra blank line at the end of the fixture files and gives a different signature (perhaps there shouldn't actually be an extra blank line at the end). Seems by default `com.google.common.io.Resources` removes blank lines at the end of a file.